src/main/java/io/cryostat/agent/triggers/TriggerParser.java

```diff
@@ -35,10 +41,50 @@ public class TriggerParser {
     private final FlightRecorderHelper flightRecorderHelper;
     private final Logger log = LoggerFactory.getLogger(getClass());
 
+    @Named(ConfigModule.CRYOSTAT_AGENT_SMART_TRIGGER_CONFIG_PATH)
+    java.nio.file.Path triggerPath;
+
     public TriggerParser(FlightRecorderHelper flightRecorderHelper) {
         this.flightRecorderHelper = flightRecorderHelper;
     }
 
+    public List<SmartTrigger> parseFromFiles() {
+        List<SmartTrigger> triggers = new ArrayList<SmartTrigger>();
+        if (!checkDir()) {
+            log.warn(
+                    "Configuration directory "
+                            + triggerPath.toString()
+                            + " doesn't exist or is missing permissions");
+            return Collections.emptyList();
+        }
+        try {
+            Files.walk(triggerPath)
+                    .filter(Files::isRegularFile)
+                    .filter(Files::isReadable)
+                    .map(path -> triggers.addAll(createFromFile(path)));
+        } catch (IOException e) {
+            log.error(e.getMessage());
+        }
+        return triggers;
+    }
+
+    private List<SmartTrigger> createFromFile(java.nio.file.Path path) {
+        try {
+            String triggerDefinitions = Files.readString(path);
+            return parse(triggerDefinitions);
```

Comment:
Since this just passes through to the existing `parse` method it expects that the file content will be just like the previous system property or environment variable value format. I think this is a missed opportunity to allow the definitions provided by files to be a little more readable, by using newline separators rather than commas. Maybe a `java.util.Scanner` or something would be a useful addition here as an additional pass on the raw file contents before handing over to the `parse` method?